api/handle_transfer_alerts.go

```diff
@@ -0,0 +1,173 @@
+	var partnerId string
+	if creds.PartnerId != nil {
+		partnerId = *creds.PartnerId
+	}
+
+	var filters struct {
+		TransferId string `form:"transfer_id"`
+	}
+	if err := c.ShouldBind(&filters); err != nil {
+		c.Status(http.StatusBadRequest)
+		return
+	}
+
+	usecase := api.UsecasesWithCreds(c.Request).NewTransferAlertsUsecase()
+	alerts, err := usecase.ListTransferAlerts(
+		c.Request.Context(),
+		creds.OrganizationId,
+		partnerId,
+		"beneficiary",
+		null.NewString(filters.TransferId, filters.TransferId != ""),
+	)
+	if presentError(c, err) {
+		return
+	}
+
+	c.JSON(http.StatusOK, gin.H{"alerts": pure_utils.Map(alerts, dto.AdaptBeneficiaryTransferAlert)})
+}
+
+func (api *API) handleCreateTransferAlert(c *gin.Context) {
+	var data dto.TransferAlertCreateBody
+	if err := c.ShouldBindJSON(&data); err != nil {
+		c.Status(http.StatusBadRequest)
+		return
+	}
+
+	creds, _ := utils.CredentialsFromCtx(c.Request.Context())
+	var partnerId string
+	if creds.PartnerId != nil {
+		partnerId = *creds.PartnerId
+	}
+
+	usecase := api.UsecasesWithCreds(c.Request).NewTransferAlertsUsecase()
+
+	alert, err := usecase.CreateTransferAlert(c.Request.Context(), models.TransferAlertCreateBody{
+		TransferId:         data.TransferId,
+		OrganizationId:     creds.OrganizationId,
+		SenderPartnerId:    partnerId,
+		Message:            data.Message,
+		TransferEndToEndId: data.TransferEndToEndId,
+		BeneficiaryIban:    data.BeneficiaryIban,
+		SenderIban:         data.SenderIban,
+	})
+	if presentError(c, err) {
+		return
+	}
+
+	c.JSON(http.StatusOK, gin.H{"alert": dto.AdaptSenderTransferAlert(alert)})
+}
+
+func (api *API) handleUpdateTransferAlertSender(c *gin.Context) {
+	alertId := c.Param("alert_id")
+	creds, _ := utils.CredentialsFromCtx(c.Request.Context())
+
+	var data dto.TransferAlertUpdateBody
+	if err := c.ShouldBindJSON(&data); err != nil {
+		c.Status(http.StatusBadRequest)
+		return
+	}
+
+	usecase := api.UsecasesWithCreds(c.Request).NewTransferAlertsUsecase()
+	alert, err := usecase.UpdateTransferAlertAsSender(c.Request.Context(), alertId, models.TransferAlertUpdateBodySender{
+		Message:            data.Message,
+		TransferEndToEndId: data.TransferEndToEndId,
+		BeneficiaryIban:    data.BeneficiaryIban,
+		SenderIban:         data.SenderIban,
+	}, creds.OrganizationId)
+	if presentError(c, err) {
+		return
+	}
+
+	c.JSON(http.StatusOK, gin.H{"alert": dto.AdaptSenderTransferAlert(alert)})
+}
+
+func (api *API) handleUpdateTransferAlertBeneficiary(c *gin.Context) {
+	alertId := c.Param("alert_id")
+	creds, _ := utils.CredentialsFromCtx(c.Request.Context())
+
+	var data dto.TransferAlertUpdateBody
+	if err := c.ShouldBindJSON(&data); err != nil {
+		c.Status(http.StatusBadRequest)
+		return
+	}
+
+	usecase := api.UsecasesWithCreds(c.Request).NewTransferAlertsUsecase()
+	alert, err := usecase.UpdateTransferAlertAsBeneficiary(c.Request.Context(), alertId, models.TransferAlertUpdateBodyBeneficiary{
+		Status: data.Status,
+	}, creds.OrganizationId)
```

Comment:
I would prefer two distinct `dto`.
If the models are distinct, I expect dtos to be distinct too. (I look at dtos + routes + handlers as the source for openapi spec).

Reply:
Alright, I don't mind adding a second one.
I'll do that.